python/lsst/galaxy_shear/processShearTest.py

```diff
@@ -0,0 +1,144 @@
+    def buildSourceCatalog(self, imageBBox, dataRef):
+        """Build an empty source catalog, using the provided sim catalog's position to generate
+        square Footprints and its ID to set that of the source catalog.
+        """
+        sourceCat = lsst.afw.table.SourceCatalog(self.schema)
+        sourceCat.getTable().setMetadata(self.algMetadata)
+        simCat = dataRef.get("epoch_catalog", immediate=True)
+        xKey = simCat.schema.find('x').key
+        yKey = simCat.schema.find('y').key
+        idKey = simCat.schema.find('ID').key
+        nGals = imageBBox.getWidth() / self.config.galaxyStampSize
+        assert nGals * self.config.galaxyStampSize == imageBBox.getWidth()
+        n = imageBBox.getWidth() / nGals
+        self.dims = lsst.afw.geom.Extent2I(n, n)
+        offset = lsst.afw.geom.Extent2I(simCat[0][xKey], simCat[0][yKey])
+        for simRecord in simCat:
+            sourceRecord = sourceCat.addNew()
+            sourceRecord.setId(simRecord.get(idKey))
+            sourceRecord.set("psf_index", simRecord.get("psf_index"))
+            sourceRecord.set("psf_library", simRecord.get("psf_library"))
+            position = lsst.afw.geom.Point2I(simRecord.get(xKey), simRecord.get(yKey))
+            bbox = lsst.afw.geom.Box2I(position - offset, self.dims)
+            footprint = lsst.afw.detection.Footprint(bbox, imageBBox)
+            peakRecord = footprint.getPeaks().addNew()
+            peakRecord.setFx(position.getX())
+            peakRecord.setFy(position.getY())
+            sourceRecord.setFootprint(footprint)
+        return sourceCat
+
+    #   Run a measurement algorithm on the set of sources provided by
+    #   a GalSim catalog.  Also run a centroid algorithm if one is provided
+    def run(self, dataRef):
+        exposure = self.buildExposure(dataRef)
+        sourceCat = self.buildSourceCatalog(exposure.getBBox(lsst.afw.image.PARENT), dataRef)
+        images_file = dataRef.get("image", immediate=True)
+        #   the GalSim catalog will provide the number of the psf_library
+        #   and the index of the hdu with the psf for each source.
+        for source in sourceCat:
+            #   Locate the psb_library and get the indexed psf
+            dataRef.dataId["psf_index"] = source.get("psf_index")
+            dataRef.dataId["psf_library"] = source.get("psf_library")
+            psf_file = dataRef.get("psf_library", immediate=True)
+            data = psf_file.getArray().astype(numpy.float64)
+            kernel = lsst.afw.math.FixedKernel(lsst.afw.image.ImageD(data))
+            psf = lsst.meas.algorithms.KernelPsf(kernel)
+            # Create a bounding box around the galaxy image of self.dims
+            x = int(source.getFootprint().getCentroid().getX()+1) - (self.dims.getX()/2)
+            y = int(source.getFootprint().getCentroid().getY()+1) - (self.dims.getY()/2)
+            bbox = lsst.afw.geom.Box2I(lsst.afw.geom.Point2I(x,y), self.dims)
```

Comment:
Probably want to use `//` here for integer division.

By the way, this is fine, but a trick I like for creating a Box2I centered around a point is to create an empty `Box2I` at the center point, then use `grow()`.  That guarantees an odd-sized box, though, and that may not be what you want here.